src/lib/core/CHIPTLVWriter.cpp

```diff
@@ -932,9 +927,9 @@ CHIP_ERROR TLVWriter::VPutStringF(uint64_t tag, const char * fmt, va_list ap)
 
     va_end(aq);
 
-#elif HAVE_MALLOC
+#else // CONFIG_HAVE_VSNPRINTF_EX
 
-    tmpBuf = static_cast<char *>(malloc(dataLen + 1));
+    tmpBuf = static_cast<char *>(chip::Platform::MemoryAlloc(dataLen + 1));
```

Comment:
Why is this write to a temp buffer better than the old `!HAVE_MALLOC` code which directly wrote it into our buffer?  Especially if we fix that code to properly length-check how much space we have left...

Reply:
It isn't, but my intent is for there to be no changes in this PR beyond what's required for a straight substitution of malloc→MemoryAlloc() etc. Later PRs can try to eliminate some of those uses entirely.